From 2dab76490d8d9b30119204a6376e00176fad9b07 Mon Sep 17 00:00:00 2001
From: Ryan Eatmon <reatmon@ti.com>
Date: Mon, 29 Dec 2025 12:42:29 -0600
Subject: [PATCH] listmount04: Update for 6.18 kernel headers

The spare slot is now being used as of the 6.18 kernel.

Upstream-Status: Pending

Signed-off-by: Ryan Eatmon <reatmon@ti.com>
---
 testcases/kernel/syscalls/listmount/listmount04.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/testcases/kernel/syscalls/listmount/listmount04.c b/testcases/kernel/syscalls/listmount/listmount04.c
index d38408d2c..2053532f6 100644
--- a/testcases/kernel/syscalls/listmount/listmount04.c
+++ b/testcases/kernel/syscalls/listmount/listmount04.c
@@ -28,7 +28,7 @@ static uint64_t mnt_ids[MNT_SIZE];
 static struct tcase {
 	int req_usage;
 	uint32_t size;
-	uint32_t spare;
+	uint32_t mnt_ns_fd;
 	uint64_t mnt_id;
 	uint64_t param;
 	uint64_t *mnt_ids;
@@ -75,12 +75,12 @@ static struct tcase {
 	{
 		.req_usage = 1,
 		.size = MNT_ID_REQ_SIZE_VER0,
-		.spare = -1,
+		.mnt_ns_fd = -1,
 		.mnt_id = LSMT_ROOT,
 		.mnt_ids = mnt_ids,
 		.nr_mnt_ids = MNT_SIZE,
 		.exp_errno = EINVAL,
-		.msg = "invalid mnt_id_req.spare",
+		.msg = "invalid mnt_id_req.mnt_ns_fd",
 	},
 	{
 		.req_usage = 1,
@@ -124,7 +124,7 @@ static void run(unsigned int n)
 		req->mnt_id = tc->mnt_id;
 		req->param = tc->param;
 		req->size = tc->size;
-		req->spare = tc->spare;
+		req->mnt_ns_fd = tc->mnt_ns_fd;
 	}
 
 	TST_EXP_FAIL(tst_syscall(__NR_listmount, req, tc->mnt_ids,
-- 
2.43.0

